spring-integration-test/src/main/java/org/springframework/integration/test/context/SpringIntegrationTestExecutionListener.java

```diff
@@ -49,16 +49,17 @@ public void prepareTestInstance(TestContext testContext) {
 		ApplicationContext applicationContext = testContext.getApplicationContext();
 		MockIntegrationContext mockIntegrationContext = applicationContext.getBean(MockIntegrationContext.class);
 		this.autoStartupCandidates = mockIntegrationContext.getAutoStartupCandidates();
-		this.autoStartupCandidates
-				.stream()
+		this.autoStartupCandidates.stream()
 				.filter(endpoint -> !match(endpoint.getBeanName(), patterns))
 				.peek(endpoint -> endpoint.setAutoStartup(true))
 				.forEach(AbstractEndpoint::start);
 	}
 
 	@Override
 	public void afterTestClass(TestContext testContext) {
-		this.autoStartupCandidates.forEach(AbstractEndpoint::stop);
+		this.autoStartupCandidates.stream()
+				.peek(endpoint -> endpoint.setAutoStartup(false))
```

Comment:
Peek is usually used for debugging purposes. (IIRC) 

So should we rewrite the stream as follows:
```java
	@Override
	public void afterTestClass(TestContext testContext) {
		this.autoStartupCandidates.forEach(abstractEndpoint -> {
					abstractEndpoint.setAutoStartup(false);
					abstractEndpoint.stop();
				});
	}
```